orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -506,6 +508,53 @@ IMPLEMENT_GRADIENT_BUILDER(GetGatherNDGradient) {
 };
 
 IMPLEMENT_GRADIENT_BUILDER(GetReshapeGradient) {
+  std::vector<Dimension> target_shape;
+  std::vector<Dimension> source_shape;
+  if (GetShape(I(0), target_shape).IsOK() &&
```

Comment:
>GetShape(I(0), target_shape).IsOK() [](start = 6, length = 35)

if source have a concrete shape, we can also skip the Shape() node

Reply:
Do you mean target_shape? Here source means the source of Shape op. If target_shape is all static, then constant folding should be able to get rid of Shape op.

---
In reply to: [465863179](https://github.com/microsoft/onnxruntime/pull/4708#discussion_r465863179) [](ancestors = 465863179)